pkg/ccl/sqlproxyccl/forwarder.go

```diff
@@ -26,6 +26,10 @@ import (
 // considered idle if no messages have been received from / sent to the client.
 const idleTimeout = 30 * time.Second
 
+// errForwarderStarted will be returned when Run is called more than once to
+// denote that the forwarder has already been started.
+var errForwarderStarted = errors.New("forwarder has already been started")
```

Comment:
nit: errForwarderStarted does not need to be defined as a constant. It can be defined inline in run. IMO errors should only be defined as a constant if:
1. Something (other than a test) needs to check the error using errors.Is.
2. In performance critical code, returning a constant error can prevent an allocation. This should only be used if the error is expected to show up in the serving path. Run is in the critical path, but this error is more of an assertion than an expected error case.